File: project/rest/domain/College.go
package domain

import (
	"fmt"
	"io/ioutil"
	"net/http"
)

/*
Name
Address
City
Zip
Web
ProgramName
 */

type Colleges []College

type College struct {

	 Name string `json:"name" bson:"name"`
	 Address string `json:"address" bson:"address"`
	 City string `json:"city" bson:"city"`
	 Zip string  `json:"zip" bson:"zip"`
	 Web string  `json:"web" bson:"web"`
	 ProgramName string `json:"programName" bson:"programName"`

}

func (c College) GetData() string {
	file, err := ioutil.ReadFile("./data/Colleges.json")
	if err != nil {
		return ""
	}
	rest := string([]byte(file))
	return  rest
}

func (c College) ServeData(w http.ResponseWriter, r *http.Request){
	w.Header().Set("Access-Control-Allow-Origin", "*")

	w.Header().Set("Access-Control-Allow-Headers", "Content-Type")

	fmt.Fprintln(w,College{}.GetData())
}

File: project/client/src/components/model/College.js
import React,{Component} from "react";
import axios from "axios";

export default class College extends Component {

    state = {college:[],loaded:0};

    componentDidMount() {

        /**
         * {Address, City, Name, ProgramName, Web, Zip}
         */
        axios.get("/college")
            .then(res => {
                const college = res.data;
                this.setState({college:college,loaded:1});
             })
            .catch(err => console.log("No Data"));
    }

    render(){
        return (
            <div>
                <h1>Loaded: {this.state.loaded}</h1>
            </div>
        );
    }

}